// @frankycty/rm-extra/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "pickWhen", {
  enumerable: true,
  get: function get() {
    return _pickWhen.default;
  }
});
Object.defineProperty(exports, "pickThenRename", {
  enumerable: true,
  get: function get() {
    return _pickThenRename.default;
  }
});
Object.defineProperty(exports, "F", {
  enumerable: true,
  get: function get() {
    return _ramda.F;
  }
});
Object.defineProperty(exports, "T", {
  enumerable: true,
  get: function get() {
    return _ramda.T;
  }
});
Object.defineProperty(exports, "__", {
  enumerable: true,
  get: function get() {
    return _ramda.__;
  }
});
Object.defineProperty(exports, "add", {
  enumerable: true,
  get: function get() {
    return _ramda.add;
  }
});
Object.defineProperty(exports, "addIndex", {
  enumerable: true,
  get: function get() {
    return _ramda.addIndex;
  }
});
Object.defineProperty(exports, "adjust", {
  enumerable: true,
  get: function get() {
    return _ramda.adjust;
  }
});
Object.defineProperty(exports, "all", {
  enumerable: true,
  get: function get() {
    return _ramda.all;
  }
});
Object.defineProperty(exports, "allPass", {
  enumerable: true,
  get: function get() {
    return _ramda.allPass;
  }
});
Object.defineProperty(exports, "always", {
  enumerable: true,
  get: function get() {
    return _ramda.always;
  }
});
Object.defineProperty(exports, "and", {
  enumerable: true,
  get: function get() {
    return _ramda.and;
  }
});
Object.defineProperty(exports, "any", {
  enumerable: true,
  get: function get() {
    return _ramda.any;
  }
});
Object.defineProperty(exports, "anyPass", {
  enumerable: true,
  get: function get() {
    return _ramda.anyPass;
  }
});
Object.defineProperty(exports, "ap", {
  enumerable: true,
  get: function get() {
    return _ramda.ap;
  }
});
Object.defineProperty(exports, "aperture", {
  enumerable: true,
  get: function get() {
    return _ramda.aperture;
  }
});
Object.defineProperty(exports, "append", {
  enumerable: true,
  get: function get() {
    return _ramda.append;
  }
});
Object.defineProperty(exports, "apply", {
  enumerable: true,
  get: function get() {
    return _ramda.apply;
  }
});
Object.defineProperty(exports, "applySpec", {
  enumerable: true,
  get: function get() {
    return _ramda.applySpec;
  }
});
Object.defineProperty(exports, "applyTo", {
  enumerable: true,
  get: function get() {
    return _ramda.applyTo;
  }
});
Object.defineProperty(exports, "ascend", {
  enumerable: true,
  get: function get() {
    return _ramda.ascend;
  }
});
Object.defineProperty(exports, "assoc", {
  enumerable: true,
  get: function get() {
    return _ramda.assoc;
  }
});
Object.defineProperty(exports, "assocPath", {
  enumerable: true,
  get: function get() {
    return _ramda.assocPath;
  }
});
Object.defineProperty(exports, "binary", {
  enumerable: true,
  get: function get() {
    return _ramda.binary;
  }
});
Object.defineProperty(exports, "bind", {
  enumerable: true,
  get: function get() {
    return _ramda.bind;
  }
});
Object.defineProperty(exports, "both", {
  enumerable: true,
  get: function get() {
    return _ramda.both;
  }
});
Object.defineProperty(exports, "call", {
  enumerable: true,
  get: function get() {
    return _ramda.call;
  }
});
Object.defineProperty(exports, "chain", {
  enumerable: true,
  get: function get() {
    return _ramda.chain;
  }
});
Object.defineProperty(exports, "clamp", {
  enumerable: true,
  get: function get() {
    return _ramda.clamp;
  }
});
Object.defineProperty(exports, "clone", {
  enumerable: true,
  get: function get() {
    return _ramda.clone;
  }
});
Object.defineProperty(exports, "comparator", {
  enumerable: true,
  get: function get() {
    return _ramda.comparator;
  }
});
Object.defineProperty(exports, "complement", {
  enumerable: true,
  get: function get() {
    return _ramda.complement;
  }
});
Object.defineProperty(exports, "compose", {
  enumerable: true,
  get: function get() {
    return _ramda.compose;
  }
});
Object.defineProperty(exports, "composeK", {
  enumerable: true,
  get: function get() {
    return _ramda.composeK;
  }
});
Object.defineProperty(exports, "composeP", {
  enumerable: true,
  get: function get() {
    return _ramda.composeP;
  }
});
Object.defineProperty(exports, "composeWith", {
  enumerable: true,
  get: function get() {
    return _ramda.composeWith;
  }
});
Object.defineProperty(exports, "concat", {
  enumerable: true,
  get: function get() {
    return _ramda.concat;
  }
});
Object.defineProperty(exports, "cond", {
  enumerable: true,
  get: function get() {
    return _ramda.cond;
  }
});
Object.defineProperty(exports, "construct", {
  enumerable: true,
  get: function get() {
    return _ramda.construct;
  }
});
Object.defineProperty(exports, "constructN", {
  enumerable: true,
  get: function get() {
    return _ramda.constructN;
  }
});
Object.defineProperty(exports, "contains", {
  enumerable: true,
  get: function get() {
    return _ramda.contains;
  }
});
Object.defineProperty(exports, "converge", {
  enumerable: true,
  get: function get() {
    return _ramda.converge;
  }
});
Object.defineProperty(exports, "countBy", {
  enumerable: true,
  get: function get() {
    return _ramda.countBy;
  }
});
Object.defineProperty(exports, "curry", {
  enumerable: true,
  get: function get() {
    return _ramda.curry;
  }
});
Object.defineProperty(exports, "curryN", {
  enumerable: true,
  get: function get() {
    return _ramda.curryN;
  }
});
Object.defineProperty(exports, "dec", {
  enumerable: true,
  get: function get() {
    return _ramda.dec;
  }
});
Object.defineProperty(exports, "defaultTo", {
  enumerable: true,
  get: function get() {
    return _ramda.defaultTo;
  }
});
Object.defineProperty(exports, "descend", {
  enumerable: true,
  get: function get() {
    return _ramda.descend;
  }
});
Object.defineProperty(exports, "difference", {
  enumerable: true,
  get: function get() {
    return _ramda.difference;
  }
});
Object.defineProperty(exports, "differenceWith", {
  enumerable: true,
  get: function get() {
    return _ramda.differenceWith;
  }
});
Object.defineProperty(exports, "dissoc", {
  enumerable: true,
  get: function get() {
    return _ramda.dissoc;
  }
});
Object.defineProperty(exports, "dissocPath", {
  enumerable: true,
  get: function get() {
    return _ramda.dissocPath;
  }
});
Object.defineProperty(exports, "divide", {
  enumerable: true,
  get: function get() {
    return _ramda.divide;
  }
});
Object.defineProperty(exports, "drop", {
  enumerable: true,
  get: function get() {
    return _ramda.drop;
  }
});
Object.defineProperty(exports, "dropLast", {
  enumerable: true,
  get: function get() {
    return _ramda.dropLast;
  }
});
Object.defineProperty(exports, "dropLastWhile", {
  enumerable: true,
  get: function get() {
    return _ramda.dropLastWhile;
  }
});
Object.defineProperty(exports, "dropRepeats", {
  enumerable: true,
  get: function get() {
    return _ramda.dropRepeats;
  }
});
Object.defineProperty(exports, "dropRepeatsWith", {
  enumerable: true,
  get: function get() {
    return _ramda.dropRepeatsWith;
  }
});
Object.defineProperty(exports, "dropWhile", {
  enumerable: true,
  get: function get() {
    return _ramda.dropWhile;
  }
});
Object.defineProperty(exports, "either", {
  enumerable: true,
  get: function get() {
    return _ramda.either;
  }
});
Object.defineProperty(exports, "empty", {
  enumerable: true,
  get: function get() {
    return _ramda.empty;
  }
});
Object.defineProperty(exports, "endsWith", {
  enumerable: true,
  get: function get() {
    return _ramda.endsWith;
  }
});
Object.defineProperty(exports, "eqBy", {
  enumerable: true,
  get: function get() {
    return _ramda.eqBy;
  }
});
Object.defineProperty(exports, "eqProps", {
  enumerable: true,
  get: function get() {
    return _ramda.eqProps;
  }
});
Object.defineProperty(exports, "equals", {
  enumerable: true,
  get: function get() {
    return _ramda.equals;
  }
});
Object.defineProperty(exports, "evolve", {
  enumerable: true,
  get: function get() {
    return _ramda.evolve;
  }
});
Object.defineProperty(exports, "filter", {
  enumerable: true,
  get: function get() {
    return _ramda.filter;
  }
});
Object.defineProperty(exports, "find", {
  enumerable: true,
  get: function get() {
    return _ramda.find;
  }
});
Object.defineProperty(exports, "findIndex", {
  enumerable: true,
  get: function get() {
    return _ramda.findIndex;
  }
});
Object.defineProperty(exports, "findLast", {
  enumerable: true,
  get: function get() {
    return _ramda.findLast;
  }
});
Object.defineProperty(exports, "findLastIndex", {
  enumerable: true,
  get: function get() {
    return _ramda.findLastIndex;
  }
});
Object.defineProperty(exports, "flatten", {
  enumerable: true,
  get: function get() {
    return _ramda.flatten;
  }
});
Object.defineProperty(exports, "flip", {
  enumerable: true,
  get: function get() {
    return _ramda.flip;
  }
});
Object.defineProperty(exports, "forEach", {
  enumerable: true,
  get: function get() {
    return _ramda.forEach;
  }
});
Object.defineProperty(exports, "forEachObjIndexed", {
  enumerable: true,
  get: function get() {
    return _ramda.forEachObjIndexed;
  }
});
Object.defineProperty(exports, "fromPairs", {
  enumerable: true,
  get: function get() {
    return _ramda.fromPairs;
  }
});
Object.defineProperty(exports, "groupBy", {
  enumerable: true,
  get: function get() {
    return _ramda.groupBy;
  }
});
Object.defineProperty(exports, "groupWith", {
  enumerable: true,
  get: function get() {
    return _ramda.groupWith;
  }
});
Object.defineProperty(exports, "gt", {
  enumerable: true,
  get: function get() {
    return _ramda.gt;
  }
});
Object.defineProperty(exports, "gte", {
  enumerable: true,
  get: function get() {
    return _ramda.gte;
  }
});
Object.defineProperty(exports, "has", {
  enumerable: true,
  get: function get() {
    return _ramda.has;
  }
});
Object.defineProperty(exports, "hasIn", {
  enumerable: true,
  get: function get() {
    return _ramda.hasIn;
  }
});
Object.defineProperty(exports, "hasPath", {
  enumerable: true,
  get: function get() {
    return _ramda.hasPath;
  }
});
Object.defineProperty(exports, "head", {
  enumerable: true,
  get: function get() {
    return _ramda.head;
  }
});
Object.defineProperty(exports, "identical", {
  enumerable: true,
  get: function get() {
    return _ramda.identical;
  }
});
Object.defineProperty(exports, "identity", {
  enumerable: true,
  get: function get() {
    return _ramda.identity;
  }
});
Object.defineProperty(exports, "ifElse", {
  enumerable: true,
  get: function get() {
    return _ramda.ifElse;
  }
});
Object.defineProperty(exports, "inc", {
  enumerable: true,
  get: function get() {
    return _ramda.inc;
  }
});
Object.defineProperty(exports, "includes", {
  enumerable: true,
  get: function get() {
    return _ramda.includes;
  }
});
Object.defineProperty(exports, "indexBy", {
  enumerable: true,
  get: function get() {
    return _ramda.indexBy;
  }
});
Object.defineProperty(exports, "indexOf", {
  enumerable: true,
  get: function get() {
    return _ramda.indexOf;
  }
});
Object.defineProperty(exports, "init", {
  enumerable: true,
  get: function get() {
    return _ramda.init;
  }
});
Object.defineProperty(exports, "innerJoin", {
  enumerable: true,
  get: function get() {
    return _ramda.innerJoin;
  }
});
Object.defineProperty(exports, "insert", {
  enumerable: true,
  get: function get() {
    return _ramda.insert;
  }
});
Object.defineProperty(exports, "insertAll", {
  enumerable: true,
  get: function get() {
    return _ramda.insertAll;
  }
});
Object.defineProperty(exports, "intersection", {
  enumerable: true,
  get: function get() {
    return _ramda.intersection;
  }
});
Object.defineProperty(exports, "intersperse", {
  enumerable: true,
  get: function get() {
    return _ramda.intersperse;
  }
});
Object.defineProperty(exports, "into", {
  enumerable: true,
  get: function get() {
    return _ramda.into;
  }
});
Object.defineProperty(exports, "invert", {
  enumerable: true,
  get: function get() {
    return _ramda.invert;
  }
});
Object.defineProperty(exports, "invertObj", {
  enumerable: true,
  get: function get() {
    return _ramda.invertObj;
  }
});
Object.defineProperty(exports, "invoker", {
  enumerable: true,
  get: function get() {
    return _ramda.invoker;
  }
});
Object.defineProperty(exports, "is", {
  enumerable: true,
  get: function get() {
    return _ramda.is;
  }
});
Object.defineProperty(exports, "isEmpty", {
  enumerable: true,
  get: function get() {
    return _ramda.isEmpty;
  }
});
Object.defineProperty(exports, "isNil", {
  enumerable: true,
  get: function get() {
    return _ramda.isNil;
  }
});
Object.defineProperty(exports, "join", {
  enumerable: true,
  get: function get() {
    return _ramda.join;
  }
});
Object.defineProperty(exports, "juxt", {
  enumerable: true,
  get: function get() {
    return _ramda.juxt;
  }
});
Object.defineProperty(exports, "keys", {
  enumerable: true,
  get: function get() {
    return _ramda.keys;
  }
});
Object.defineProperty(exports, "keysIn", {
  enumerable: true,
  get: function get() {
    return _ramda.keysIn;
  }
});
Object.defineProperty(exports, "last", {
  enumerable: true,
  get: function get() {
    return _ramda.last;
  }
});
Object.defineProperty(exports, "lastIndexOf", {
  enumerable: true,
  get: function get() {
    return _ramda.lastIndexOf;
  }
});
Object.defineProperty(exports, "length", {
  enumerable: true,
  get: function get() {
    return _ramda.length;
  }
});
Object.defineProperty(exports, "lens", {
  enumerable: true,
  get: function get() {
    return _ramda.lens;
  }
});
Object.defineProperty(exports, "lensIndex", {
  enumerable: true,
  get: function get() {
    return _ramda.lensIndex;
  }
});
Object.defineProperty(exports, "lensPath", {
  enumerable: true,
  get: function get() {
    return _ramda.lensPath;
  }
});
Object.defineProperty(exports, "lensProp", {
  enumerable: true,
  get: function get() {
    return _ramda.lensProp;
  }
});
Object.defineProperty(exports, "lift", {
  enumerable: true,
  get: function get() {
    return _ramda.lift;
  }
});
Object.defineProperty(exports, "liftN", {
  enumerable: true,
  get: function get() {
    return _ramda.liftN;
  }
});
Object.defineProperty(exports, "lt", {
  enumerable: true,
  get: function get() {
    return _ramda.lt;
  }
});
Object.defineProperty(exports, "lte", {
  enumerable: true,
  get: function get() {
    return _ramda.lte;
  }
});
Object.defineProperty(exports, "map", {
  enumerable: true,
  get: function get() {
    return _ramda.map;
  }
});
Object.defineProperty(exports, "mapAccum", {
  enumerable: true,
  get: function get() {
    return _ramda.mapAccum;
  }
});
Object.defineProperty(exports, "mapAccumRight", {
  enumerable: true,
  get: function get() {
    return _ramda.mapAccumRight;
  }
});
Object.defineProperty(exports, "mapObjIndexed", {
  enumerable: true,
  get: function get() {
    return _ramda.mapObjIndexed;
  }
});
Object.defineProperty(exports, "match", {
  enumerable: true,
  get: function get() {
    return _ramda.match;
  }
});
Object.defineProperty(exports, "mathMod", {
  enumerable: true,
  get: function get() {
    return _ramda.mathMod;
  }
});
Object.defineProperty(exports, "max", {
  enumerable: true,
  get: function get() {
    return _ramda.max;
  }
});
Object.defineProperty(exports, "maxBy", {
  enumerable: true,
  get: function get() {
    return _ramda.maxBy;
  }
});
Object.defineProperty(exports, "mean", {
  enumerable: true,
  get: function get() {
    return _ramda.mean;
  }
});
Object.defineProperty(exports, "median", {
  enumerable: true,
  get: function get() {
    return _ramda.median;
  }
});
Object.defineProperty(exports, "memoizeWith", {
  enumerable: true,
  get: function get() {
    return _ramda.memoizeWith;
  }
});
Object.defineProperty(exports, "merge", {
  enumerable: true,
  get: function get() {
    return _ramda.merge;
  }
});
Object.defineProperty(exports, "mergeAll", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeAll;
  }
});
Object.defineProperty(exports, "mergeDeepLeft", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeDeepLeft;
  }
});
Object.defineProperty(exports, "mergeDeepRight", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeDeepRight;
  }
});
Object.defineProperty(exports, "mergeDeepWith", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeDeepWith;
  }
});
Object.defineProperty(exports, "mergeDeepWithKey", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeDeepWithKey;
  }
});
Object.defineProperty(exports, "mergeLeft", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeLeft;
  }
});
Object.defineProperty(exports, "mergeRight", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeRight;
  }
});
Object.defineProperty(exports, "mergeWith", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeWith;
  }
});
Object.defineProperty(exports, "mergeWithKey", {
  enumerable: true,
  get: function get() {
    return _ramda.mergeWithKey;
  }
});
Object.defineProperty(exports, "min", {
  enumerable: true,
  get: function get() {
    return _ramda.min;
  }
});
Object.defineProperty(exports, "minBy", {
  enumerable: true,
  get: function get() {
    return _ramda.minBy;
  }
});
Object.defineProperty(exports, "modulo", {
  enumerable: true,
  get: function get() {
    return _ramda.modulo;
  }
});
Object.defineProperty(exports, "move", {
  enumerable: true,
  get: function get() {
    return _ramda.move;
  }
});
Object.defineProperty(exports, "multiply", {
  enumerable: true,
  get: function get() {
    return _ramda.multiply;
  }
});
Object.defineProperty(exports, "nAry", {
  enumerable: true,
  get: function get() {
    return _ramda.nAry;
  }
});
Object.defineProperty(exports, "negate", {
  enumerable: true,
  get: function get() {
    return _ramda.negate;
  }
});
Object.defineProperty(exports, "none", {
  enumerable: true,
  get: function get() {
    return _ramda.none;
  }
});
Object.defineProperty(exports, "not", {
  enumerable: true,
  get: function get() {
    return _ramda.not;
  }
});
Object.defineProperty(exports, "nth", {
  enumerable: true,
  get: function get() {
    return _ramda.nth;
  }
});
Object.defineProperty(exports, "nthArg", {
  enumerable: true,
  get: function get() {
    return _ramda.nthArg;
  }
});
Object.defineProperty(exports, "o", {
  enumerable: true,
  get: function get() {
    return _ramda.o;
  }
});
Object.defineProperty(exports, "objOf", {
  enumerable: true,
  get: function get() {
    return _ramda.objOf;
  }
});
Object.defineProperty(exports, "of", {
  enumerable: true,
  get: function get() {
    return _ramda.of;
  }
});
Object.defineProperty(exports, "omit", {
  enumerable: true,
  get: function get() {
    return _ramda.omit;
  }
});
Object.defineProperty(exports, "once", {
  enumerable: true,
  get: function get() {
    return _ramda.once;
  }
});
Object.defineProperty(exports, "or", {
  enumerable: true,
  get: function get() {
    return _ramda.or;
  }
});
Object.defineProperty(exports, "otherwise", {
  enumerable: true,
  get: function get() {
    return _ramda.otherwise;
  }
});
Object.defineProperty(exports, "over", {
  enumerable: true,
  get: function get() {
    return _ramda.over;
  }
});
Object.defineProperty(exports, "pair", {
  enumerable: true,
  get: function get() {
    return _ramda.pair;
  }
});
Object.defineProperty(exports, "partial", {
  enumerable: true,
  get: function get() {
    return _ramda.partial;
  }
});
Object.defineProperty(exports, "partialRight", {
  enumerable: true,
  get: function get() {
    return _ramda.partialRight;
  }
});
Object.defineProperty(exports, "partition", {
  enumerable: true,
  get: function get() {
    return _ramda.partition;
  }
});
Object.defineProperty(exports, "path", {
  enumerable: true,
  get: function get() {
    return _ramda.path;
  }
});
Object.defineProperty(exports, "paths", {
  enumerable: true,
  get: function get() {
    return _ramda.paths;
  }
});
Object.defineProperty(exports, "pathEq", {
  enumerable: true,
  get: function get() {
    return _ramda.pathEq;
  }
});
Object.defineProperty(exports, "pathOr", {
  enumerable: true,
  get: function get() {
    return _ramda.pathOr;
  }
});
Object.defineProperty(exports, "pathSatisfies", {
  enumerable: true,
  get: function get() {
    return _ramda.pathSatisfies;
  }
});
Object.defineProperty(exports, "pick", {
  enumerable: true,
  get: function get() {
    return _ramda.pick;
  }
});
Object.defineProperty(exports, "pickAll", {
  enumerable: true,
  get: function get() {
    return _ramda.pickAll;
  }
});
Object.defineProperty(exports, "pickBy", {
  enumerable: true,
  get: function get() {
    return _ramda.pickBy;
  }
});
Object.defineProperty(exports, "pipe", {
  enumerable: true,
  get: function get() {
    return _ramda.pipe;
  }
});
Object.defineProperty(exports, "pipeK", {
  enumerable: true,
  get: function get() {
    return _ramda.pipeK;
  }
});
Object.defineProperty(exports, "pipeP", {
  enumerable: true,
  get: function get() {
    return _ramda.pipeP;
  }
});
Object.defineProperty(exports, "pipeWith", {
  enumerable: true,
  get: function get() {
    return _ramda.pipeWith;
  }
});
Object.defineProperty(exports, "pluck", {
  enumerable: true,
  get: function get() {
    return _ramda.pluck;
  }
});
Object.defineProperty(exports, "prepend", {
  enumerable: true,
  get: function get() {
    return _ramda.prepend;
  }
});
Object.defineProperty(exports, "product", {
  enumerable: true,
  get: function get() {
    return _ramda.product;
  }
});
Object.defineProperty(exports, "project", {
  enumerable: true,
  get: function get() {
    return _ramda.project;
  }
});
Object.defineProperty(exports, "prop", {
  enumerable: true,
  get: function get() {
    return _ramda.prop;
  }
});
Object.defineProperty(exports, "propEq", {
  enumerable: true,
  get: function get() {
    return _ramda.propEq;
  }
});
Object.defineProperty(exports, "propIs", {
  enumerable: true,
  get: function get() {
    return _ramda.propIs;
  }
});
Object.defineProperty(exports, "propOr", {
  enumerable: true,
  get: function get() {
    return _ramda.propOr;
  }
});
Object.defineProperty(exports, "propSatisfies", {
  enumerable: true,
  get: function get() {
    return _ramda.propSatisfies;
  }
});
Object.defineProperty(exports, "props", {
  enumerable: true,
  get: function get() {
    return _ramda.props;
  }
});
Object.defineProperty(exports, "range", {
  enumerable: true,
  get: function get() {
    return _ramda.range;
  }
});
Object.defineProperty(exports, "reduce", {
  enumerable: true,
  get: function get() {
    return _ramda.reduce;
  }
});
Object.defineProperty(exports, "reduceBy", {
  enumerable: true,
  get: function get() {
    return _ramda.reduceBy;
  }
});
Object.defineProperty(exports, "reduceRight", {
  enumerable: true,
  get: function get() {
    return _ramda.reduceRight;
  }
});
Object.defineProperty(exports, "reduceWhile", {
  enumerable: true,
  get: function get() {
    return _ramda.reduceWhile;
  }
});
Object.defineProperty(exports, "reduced", {
  enumerable: true,
  get: function get() {
    return _ramda.reduced;
  }
});
Object.defineProperty(exports, "reject", {
  enumerable: true,
  get: function get() {
    return _ramda.reject;
  }
});
Object.defineProperty(exports, "remove", {
  enumerable: true,
  get: function get() {
    return _ramda.remove;
  }
});
Object.defineProperty(exports, "repeat", {
  enumerable: true,
  get: function get() {
    return _ramda.repeat;
  }
});
Object.defineProperty(exports, "replace", {
  enumerable: true,
  get: function get() {
    return _ramda.replace;
  }
});
Object.defineProperty(exports, "reverse", {
  enumerable: true,
  get: function get() {
    return _ramda.reverse;
  }
});
Object.defineProperty(exports, "scan", {
  enumerable: true,
  get: function get() {
    return _ramda.scan;
  }
});
Object.defineProperty(exports, "sequence", {
  enumerable: true,
  get: function get() {
    return _ramda.sequence;
  }
});
Object.defineProperty(exports, "set", {
  enumerable: true,
  get: function get() {
    return _ramda.set;
  }
});
Object.defineProperty(exports, "slice", {
  enumerable: true,
  get: function get() {
    return _ramda.slice;
  }
});
Object.defineProperty(exports, "sort", {
  enumerable: true,
  get: function get() {
    return _ramda.sort;
  }
});
Object.defineProperty(exports, "sortBy", {
  enumerable: true,
  get: function get() {
    return _ramda.sortBy;
  }
});
Object.defineProperty(exports, "sortWith", {
  enumerable: true,
  get: function get() {
    return _ramda.sortWith;
  }
});
Object.defineProperty(exports, "split", {
  enumerable: true,
  get: function get() {
    return _ramda.split;
  }
});
Object.defineProperty(exports, "splitAt", {
  enumerable: true,
  get: function get() {
    return _ramda.splitAt;
  }
});
Object.defineProperty(exports, "splitEvery", {
  enumerable: true,
  get: function get() {
    return _ramda.splitEvery;
  }
});
Object.defineProperty(exports, "splitWhen", {
  enumerable: true,
  get: function get() {
    return _ramda.splitWhen;
  }
});
Object.defineProperty(exports, "startsWith", {
  enumerable: true,
  get: function get() {
    return _ramda.startsWith;
  }
});
Object.defineProperty(exports, "subtract", {
  enumerable: true,
  get: function get() {
    return _ramda.subtract;
  }
});
Object.defineProperty(exports, "sum", {
  enumerable: true,
  get: function get() {
    return _ramda.sum;
  }
});
Object.defineProperty(exports, "symmetricDifference", {
  enumerable: true,
  get: function get() {
    return _ramda.symmetricDifference;
  }
});
Object.defineProperty(exports, "symmetricDifferenceWith", {
  enumerable: true,
  get: function get() {
    return _ramda.symmetricDifferenceWith;
  }
});
Object.defineProperty(exports, "tail", {
  enumerable: true,
  get: function get() {
    return _ramda.tail;
  }
});
Object.defineProperty(exports, "take", {
  enumerable: true,
  get: function get() {
    return _ramda.take;
  }
});
Object.defineProperty(exports, "takeLast", {
  enumerable: true,
  get: function get() {
    return _ramda.takeLast;
  }
});
Object.defineProperty(exports, "takeLastWhile", {
  enumerable: true,
  get: function get() {
    return _ramda.takeLastWhile;
  }
});
Object.defineProperty(exports, "takeWhile", {
  enumerable: true,
  get: function get() {
    return _ramda.takeWhile;
  }
});
Object.defineProperty(exports, "tap", {
  enumerable: true,
  get: function get() {
    return _ramda.tap;
  }
});
Object.defineProperty(exports, "test", {
  enumerable: true,
  get: function get() {
    return _ramda.test;
  }
});
Object.defineProperty(exports, "andThen", {
  enumerable: true,
  get: function get() {
    return _ramda.andThen;
  }
});
Object.defineProperty(exports, "times", {
  enumerable: true,
  get: function get() {
    return _ramda.times;
  }
});
Object.defineProperty(exports, "toLower", {
  enumerable: true,
  get: function get() {
    return _ramda.toLower;
  }
});
Object.defineProperty(exports, "toPairs", {
  enumerable: true,
  get: function get() {
    return _ramda.toPairs;
  }
});
Object.defineProperty(exports, "toPairsIn", {
  enumerable: true,
  get: function get() {
    return _ramda.toPairsIn;
  }
});
Object.defineProperty(exports, "toString", {
  enumerable: true,
  get: function get() {
    return _ramda.toString;
  }
});
Object.defineProperty(exports, "toUpper", {
  enumerable: true,
  get: function get() {
    return _ramda.toUpper;
  }
});
Object.defineProperty(exports, "transduce", {
  enumerable: true,
  get: function get() {
    return _ramda.transduce;
  }
});
Object.defineProperty(exports, "transpose", {
  enumerable: true,
  get: function get() {
    return _ramda.transpose;
  }
});
Object.defineProperty(exports, "traverse", {
  enumerable: true,
  get: function get() {
    return _ramda.traverse;
  }
});
Object.defineProperty(exports, "trim", {
  enumerable: true,
  get: function get() {
    return _ramda.trim;
  }
});
Object.defineProperty(exports, "tryCatch", {
  enumerable: true,
  get: function get() {
    return _ramda.tryCatch;
  }
});
Object.defineProperty(exports, "type", {
  enumerable: true,
  get: function get() {
    return _ramda.type;
  }
});
Object.defineProperty(exports, "unapply", {
  enumerable: true,
  get: function get() {
    return _ramda.unapply;
  }
});
Object.defineProperty(exports, "unary", {
  enumerable: true,
  get: function get() {
    return _ramda.unary;
  }
});
Object.defineProperty(exports, "uncurryN", {
  enumerable: true,
  get: function get() {
    return _ramda.uncurryN;
  }
});
Object.defineProperty(exports, "unfold", {
  enumerable: true,
  get: function get() {
    return _ramda.unfold;
  }
});
Object.defineProperty(exports, "union", {
  enumerable: true,
  get: function get() {
    return _ramda.union;
  }
});
Object.defineProperty(exports, "unionWith", {
  enumerable: true,
  get: function get() {
    return _ramda.unionWith;
  }
});
Object.defineProperty(exports, "uniq", {
  enumerable: true,
  get: function get() {
    return _ramda.uniq;
  }
});
Object.defineProperty(exports, "uniqBy", {
  enumerable: true,
  get: function get() {
    return _ramda.uniqBy;
  }
});
Object.defineProperty(exports, "uniqWith", {
  enumerable: true,
  get: function get() {
    return _ramda.uniqWith;
  }
});
Object.defineProperty(exports, "unless", {
  enumerable: true,
  get: function get() {
    return _ramda.unless;
  }
});
Object.defineProperty(exports, "unnest", {
  enumerable: true,
  get: function get() {
    return _ramda.unnest;
  }
});
Object.defineProperty(exports, "until", {
  enumerable: true,
  get: function get() {
    return _ramda.until;
  }
});
Object.defineProperty(exports, "update", {
  enumerable: true,
  get: function get() {
    return _ramda.update;
  }
});
Object.defineProperty(exports, "useWith", {
  enumerable: true,
  get: function get() {
    return _ramda.useWith;
  }
});
Object.defineProperty(exports, "values", {
  enumerable: true,
  get: function get() {
    return _ramda.values;
  }
});
Object.defineProperty(exports, "valuesIn", {
  enumerable: true,
  get: function get() {
    return _ramda.valuesIn;
  }
});
Object.defineProperty(exports, "view", {
  enumerable: true,
  get: function get() {
    return _ramda.view;
  }
});
Object.defineProperty(exports, "when", {
  enumerable: true,
  get: function get() {
    return _ramda.when;
  }
});
Object.defineProperty(exports, "where", {
  enumerable: true,
  get: function get() {
    return _ramda.where;
  }
});
Object.defineProperty(exports, "whereEq", {
  enumerable: true,
  get: function get() {
    return _ramda.whereEq;
  }
});
Object.defineProperty(exports, "without", {
  enumerable: true,
  get: function get() {
    return _ramda.without;
  }
});
Object.defineProperty(exports, "xor", {
  enumerable: true,
  get: function get() {
    return _ramda.xor;
  }
});
Object.defineProperty(exports, "xprod", {
  enumerable: true,
  get: function get() {
    return _ramda.xprod;
  }
});
Object.defineProperty(exports, "zip", {
  enumerable: true,
  get: function get() {
    return _ramda.zip;
  }
});
Object.defineProperty(exports, "zipObj", {
  enumerable: true,
  get: function get() {
    return _ramda.zipObj;
  }
});
Object.defineProperty(exports, "zipWith", {
  enumerable: true,
  get: function get() {
    return _ramda.zipWith;
  }
});
Object.defineProperty(exports, "thunkify", {
  enumerable: true,
  get: function get() {
    return _ramda.thunkify;
  }
});
Object.defineProperty(exports, "isNotUndefined", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotUndefined;
  }
});
Object.defineProperty(exports, "isUndefined", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isUndefined;
  }
});
Object.defineProperty(exports, "isNull", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNull;
  }
});
Object.defineProperty(exports, "isNotNull", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotNull;
  }
});
Object.defineProperty(exports, "isNotNil", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotNil;
  }
});
Object.defineProperty(exports, "isArray", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isArray;
  }
});
Object.defineProperty(exports, "isIterable", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isIterable;
  }
});
Object.defineProperty(exports, "isEmptyArray", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isEmptyArray;
  }
});
Object.defineProperty(exports, "isNotArray", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotArray;
  }
});
Object.defineProperty(exports, "isNonEmptyArray", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNonEmptyArray;
  }
});
Object.defineProperty(exports, "isBoolean", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isBoolean;
  }
});
Object.defineProperty(exports, "isNotBoolean", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotBoolean;
  }
});
Object.defineProperty(exports, "isNilOrEmpty", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNilOrEmpty;
  }
});
Object.defineProperty(exports, "isString", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isString;
  }
});
Object.defineProperty(exports, "isEmptyString", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isEmptyString;
  }
});
Object.defineProperty(exports, "isNotString", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotString;
  }
});
Object.defineProperty(exports, "isNonEmptyString", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNonEmptyString;
  }
});
Object.defineProperty(exports, "isArrayLike", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isArrayLike;
  }
});
Object.defineProperty(exports, "isNotArrayLike", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotArrayLike;
  }
});
Object.defineProperty(exports, "isGeneratorFunction", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isGeneratorFunction;
  }
});
Object.defineProperty(exports, "isNotGeneratorFunction", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotGeneratorFunction;
  }
});
Object.defineProperty(exports, "isAsyncFunction", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isAsyncFunction;
  }
});
Object.defineProperty(exports, "isNotAsyncFunction", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotAsyncFunction;
  }
});
Object.defineProperty(exports, "isFunction", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isFunction;
  }
});
Object.defineProperty(exports, "isNotFunction", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotFunction;
  }
});
Object.defineProperty(exports, "isObj", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isObj;
  }
});
Object.defineProperty(exports, "isObject", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isObject;
  }
});
Object.defineProperty(exports, "isNotObj", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotObj;
  }
});
Object.defineProperty(exports, "isNotObject", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotObject;
  }
});
Object.defineProperty(exports, "isObjLike", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isObjLike;
  }
});
Object.defineProperty(exports, "isObjectLike", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isObjectLike;
  }
});
Object.defineProperty(exports, "isNotObjLike", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotObjLike;
  }
});
Object.defineProperty(exports, "isNotObjectLike", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotObjectLike;
  }
});
Object.defineProperty(exports, "isPlainObj", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isPlainObj;
  }
});
Object.defineProperty(exports, "isPlainObject", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isPlainObject;
  }
});
Object.defineProperty(exports, "isNotPlainObj", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotPlainObj;
  }
});
Object.defineProperty(exports, "isNotPlainObject", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotPlainObject;
  }
});
Object.defineProperty(exports, "isDate", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isDate;
  }
});
Object.defineProperty(exports, "isNotDate", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotDate;
  }
});
Object.defineProperty(exports, "isValidDate", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isValidDate;
  }
});
Object.defineProperty(exports, "isNotValidDate", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotValidDate;
  }
});
Object.defineProperty(exports, "isInvalidDate", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isInvalidDate;
  }
});
Object.defineProperty(exports, "isNumber", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNumber;
  }
});
Object.defineProperty(exports, "isNotNumber", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotNumber;
  }
});
Object.defineProperty(exports, "isPositive", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isPositive;
  }
});
Object.defineProperty(exports, "isNegative", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNegative;
  }
});
Object.defineProperty(exports, "isPositiveZero", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isPositiveZero;
  }
});
Object.defineProperty(exports, "isNegativeZero", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNegativeZero;
  }
});
Object.defineProperty(exports, "isNotNilOrEmpty", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotNilOrEmpty;
  }
});
Object.defineProperty(exports, "isNonPositive", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNonPositive;
  }
});
Object.defineProperty(exports, "isNonNegative", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNonNegative;
  }
});
Object.defineProperty(exports, "isMap", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isMap;
  }
});
Object.defineProperty(exports, "isNaN", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNaN;
  }
});
Object.defineProperty(exports, "isNotNaN", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotNaN;
  }
});
Object.defineProperty(exports, "isFinite", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isFinite;
  }
});
Object.defineProperty(exports, "isNotFinite", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotFinite;
  }
});
Object.defineProperty(exports, "isInteger", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isInteger;
  }
});
Object.defineProperty(exports, "isNotInteger", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotInteger;
  }
});
Object.defineProperty(exports, "isFloat", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isFloat;
  }
});
Object.defineProperty(exports, "isNotFloat", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotFloat;
  }
});
Object.defineProperty(exports, "isValidNumber", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isValidNumber;
  }
});
Object.defineProperty(exports, "isNotValidNumber", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotValidNumber;
  }
});
Object.defineProperty(exports, "isOdd", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isOdd;
  }
});
Object.defineProperty(exports, "isEven", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isEven;
  }
});
Object.defineProperty(exports, "isPair", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isPair;
  }
});
Object.defineProperty(exports, "isNotPair", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotPair;
  }
});
Object.defineProperty(exports, "isThenable", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isThenable;
  }
});
Object.defineProperty(exports, "isPromise", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isPromise;
  }
});
Object.defineProperty(exports, "isTrue", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isTrue;
  }
});
Object.defineProperty(exports, "isFalse", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isFalse;
  }
});
Object.defineProperty(exports, "isTruthy", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isTruthy;
  }
});
Object.defineProperty(exports, "isFalsy", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isFalsy;
  }
});
Object.defineProperty(exports, "isRegExp", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isRegExp;
  }
});
Object.defineProperty(exports, "isNotRegExp", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotRegExp;
  }
});
Object.defineProperty(exports, "isSet", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isSet;
  }
});
Object.defineProperty(exports, "isSparseArray", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isSparseArray;
  }
});
Object.defineProperty(exports, "isSymbol", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isSymbol;
  }
});
Object.defineProperty(exports, "isSafeInteger", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isSafeInteger;
  }
});
Object.defineProperty(exports, "isIndexed", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isIndexed;
  }
});
Object.defineProperty(exports, "stubUndefined", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.stubUndefined;
  }
});
Object.defineProperty(exports, "stubNull", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.stubNull;
  }
});
Object.defineProperty(exports, "stubObj", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.stubObj;
  }
});
Object.defineProperty(exports, "stubObject", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.stubObject;
  }
});
Object.defineProperty(exports, "stubString", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.stubString;
  }
});
Object.defineProperty(exports, "stubArray", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.stubArray;
  }
});
Object.defineProperty(exports, "noop", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.noop;
  }
});
Object.defineProperty(exports, "liftFN", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.liftFN;
  }
});
Object.defineProperty(exports, "liftF", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.liftF;
  }
});
Object.defineProperty(exports, "cata", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.cata;
  }
});
Object.defineProperty(exports, "weave", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.weave;
  }
});
Object.defineProperty(exports, "weaveLazy", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.weaveLazy;
  }
});
Object.defineProperty(exports, "curryRightN", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.curryRightN;
  }
});
Object.defineProperty(exports, "curryRight", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.curryRight;
  }
});
Object.defineProperty(exports, "allP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.allP;
  }
});
Object.defineProperty(exports, "noneP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.noneP;
  }
});
Object.defineProperty(exports, "resolveP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.resolveP;
  }
});
Object.defineProperty(exports, "rejectP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.rejectP;
  }
});
Object.defineProperty(exports, "delayP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.delayP;
  }
});
Object.defineProperty(exports, "thenP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.thenP;
  }
});
Object.defineProperty(exports, "then", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.then;
  }
});
Object.defineProperty(exports, "allSettledP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.allSettledP;
  }
});
Object.defineProperty(exports, "Y", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.Y;
  }
});
Object.defineProperty(exports, "seq", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.seq;
  }
});
Object.defineProperty(exports, "sequencing", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.sequencing;
  }
});
Object.defineProperty(exports, "dispatch", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.dispatch;
  }
});
Object.defineProperty(exports, "async", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.async;
  }
});
Object.defineProperty(exports, "anyP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.anyP;
  }
});
Object.defineProperty(exports, "firstP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.firstP;
  }
});
Object.defineProperty(exports, "lastP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lastP;
  }
});
Object.defineProperty(exports, "fnull", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.fnull;
  }
});
Object.defineProperty(exports, "mapIndexed", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.mapIndexed;
  }
});
Object.defineProperty(exports, "reduceIndexed", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.reduceIndexed;
  }
});
Object.defineProperty(exports, "pickIndexes", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.pickIndexes;
  }
});
Object.defineProperty(exports, "list", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.list;
  }
});
Object.defineProperty(exports, "ensureArray", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.ensureArray;
  }
});
Object.defineProperty(exports, "concatAll", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.concatAll;
  }
});
Object.defineProperty(exports, "concatRight", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.concatRight;
  }
});
Object.defineProperty(exports, "reduceP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.reduceP;
  }
});
Object.defineProperty(exports, "reduceRightP", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.reduceRightP;
  }
});
Object.defineProperty(exports, "sliceFrom", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.sliceFrom;
  }
});
Object.defineProperty(exports, "sliceTo", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.sliceTo;
  }
});
Object.defineProperty(exports, "omitIndexes", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.omitIndexes;
  }
});
Object.defineProperty(exports, "compact", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.compact;
  }
});
Object.defineProperty(exports, "appendFlipped", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.appendFlipped;
  }
});
Object.defineProperty(exports, "contained", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.contained;
  }
});
Object.defineProperty(exports, "included", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.included;
  }
});
Object.defineProperty(exports, "lengthGt", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lengthGt;
  }
});
Object.defineProperty(exports, "lengthLt", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lengthLt;
  }
});
Object.defineProperty(exports, "lengthGte", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lengthGte;
  }
});
Object.defineProperty(exports, "lengthLte", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lengthLte;
  }
});
Object.defineProperty(exports, "lengthEq", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lengthEq;
  }
});
Object.defineProperty(exports, "lengthNotEq", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lengthNotEq;
  }
});
Object.defineProperty(exports, "allEqual", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.allEqual;
  }
});
Object.defineProperty(exports, "repeatStr", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.repeatStr;
  }
});
Object.defineProperty(exports, "allIdentical", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.allIdentical;
  }
});
Object.defineProperty(exports, "allIdenticalTo", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.allIdenticalTo;
  }
});
Object.defineProperty(exports, "allEqualTo", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.allEqualTo;
  }
});
Object.defineProperty(exports, "flattenDepth", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.flattenDepth;
  }
});
Object.defineProperty(exports, "toArray", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.toArray;
  }
});
Object.defineProperty(exports, "allUnique", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.allUnique;
  }
});
Object.defineProperty(exports, "notAllUnique", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.notAllUnique;
  }
});
Object.defineProperty(exports, "sortByProps", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.sortByProps;
  }
});
Object.defineProperty(exports, "skipTake", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.skipTake;
  }
});
Object.defineProperty(exports, "renameKeys", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.renameKeys;
  }
});
Object.defineProperty(exports, "renameKeysWith", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.renameKeysWith;
  }
});
Object.defineProperty(exports, "resetToDefault", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.resetToDefault;
  }
});
Object.defineProperty(exports, "mergeProps", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.mergeProps;
  }
});
Object.defineProperty(exports, "mergePaths", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.mergePaths;
  }
});
Object.defineProperty(exports, "mergeProp", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.mergeProp;
  }
});
Object.defineProperty(exports, "mergePath", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.mergePath;
  }
});
Object.defineProperty(exports, "omitBy", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.omitBy;
  }
});
Object.defineProperty(exports, "pathOrLazy", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.pathOrLazy;
  }
});
Object.defineProperty(exports, "viewOr", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.viewOr;
  }
});
Object.defineProperty(exports, "spreadProp", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.spreadProp;
  }
});
Object.defineProperty(exports, "spreadPath", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.spreadPath;
  }
});
Object.defineProperty(exports, "flattenProp", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.flattenProp;
  }
});
Object.defineProperty(exports, "flattenPath", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.flattenPath;
  }
});
Object.defineProperty(exports, "unzipObjWith", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.unzipObjWith;
  }
});
Object.defineProperty(exports, "zipObjWith", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.zipObjWith;
  }
});
Object.defineProperty(exports, "lensEq", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lensEq;
  }
});
Object.defineProperty(exports, "lensNotEq", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lensNotEq;
  }
});
Object.defineProperty(exports, "lensSatisfies", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lensSatisfies;
  }
});
Object.defineProperty(exports, "lensNotSatisfy", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lensNotSatisfy;
  }
});
Object.defineProperty(exports, "lensTraverse", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lensTraverse;
  }
});
Object.defineProperty(exports, "lensIso", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.lensIso;
  }
});
Object.defineProperty(exports, "propNotEq", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.propNotEq;
  }
});
Object.defineProperty(exports, "pathNotEq", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.pathNotEq;
  }
});
Object.defineProperty(exports, "inRange", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.inRange;
  }
});
Object.defineProperty(exports, "isNotEmpty", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.isNotEmpty;
  }
});
Object.defineProperty(exports, "defaultWhen", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.defaultWhen;
  }
});
Object.defineProperty(exports, "notBoth", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.notBoth;
  }
});
Object.defineProperty(exports, "nand", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.nand;
  }
});
Object.defineProperty(exports, "neither", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.neither;
  }
});
Object.defineProperty(exports, "nor", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.nor;
  }
});
Object.defineProperty(exports, "notAllPass", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.notAllPass;
  }
});
Object.defineProperty(exports, "nonePass", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.nonePass;
  }
});
Object.defineProperty(exports, "argsPass", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.argsPass;
  }
});
Object.defineProperty(exports, "dropArgs", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.dropArgs;
  }
});
Object.defineProperty(exports, "round", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.round;
  }
});
Object.defineProperty(exports, "ceil", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.ceil;
  }
});
Object.defineProperty(exports, "divideNum", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.divideNum;
  }
});
Object.defineProperty(exports, "floor", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.floor;
  }
});
Object.defineProperty(exports, "trunc", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.trunc;
  }
});
Object.defineProperty(exports, "sign", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.sign;
  }
});
Object.defineProperty(exports, "subtractNum", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.subtractNum;
  }
});
Object.defineProperty(exports, "replaceAll", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.replaceAll;
  }
});
Object.defineProperty(exports, "escapeRegExp", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.escapeRegExp;
  }
});
Object.defineProperty(exports, "trimStart", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.trimStart;
  }
});
Object.defineProperty(exports, "trimLeft", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.trimLeft;
  }
});
Object.defineProperty(exports, "trimEnd", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.trimEnd;
  }
});
Object.defineProperty(exports, "trimRight", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.trimRight;
  }
});
Object.defineProperty(exports, "trimCharsEnd", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.trimCharsEnd;
  }
});
Object.defineProperty(exports, "trimCharsStart", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.trimCharsStart;
  }
});
Object.defineProperty(exports, "padCharsStart", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.padCharsStart;
  }
});
Object.defineProperty(exports, "padCharsEnd", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.padCharsEnd;
  }
});
Object.defineProperty(exports, "padEnd", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.padEnd;
  }
});
Object.defineProperty(exports, "padStart", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.padStart;
  }
});
Object.defineProperty(exports, "Identity", {
  enumerable: true,
  get: function get() {
    return _ramdaAdjunct.Identity;
  }
});

var _pickWhen = _interopRequireDefault(require("./pickWhen"));

var _pickThenRename = _interopRequireDefault(require("./pickThenRename"));

var _ramda = require("ramda");

var _ramdaAdjunct = require("ramda-adjunct");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
