// @frankycty/node-framework/lib/middlewares/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _koaCompose = _interopRequireDefault(require("koa-compose"));

var _koaHelmet = _interopRequireDefault(require("koa-helmet"));

var _koaBodyparser = _interopRequireDefault(require("koa-bodyparser"));

var _koaCompress = _interopRequireDefault(require("koa-compress"));

var _cors = _interopRequireDefault(require("./cors"));

var _errorHandling = _interopRequireDefault(require("./errorHandling"));

var _dataCache = _interopRequireDefault(require("./dataCache"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var defaultOpts = {
  bodyParserOpts: {
    enableTypes: ["json"]
  },
  loggingOpts: {
    autoLogging: false
  },
  helmetOpts: {},
  compressOpts: {},
  errorHandlingOpts: {},
  requestIdOpts: {
    expose: "Request-Id",
    header: "Request-Id"
  },
  dataCacheOpts: {}
};

var _default = () => {
  var {
    bodyParserOpts,
    helmetOpts,
    compressOpts,
    errorHandlingOpts,
    dataCacheOpts
  } = defaultOpts;
  return (0, _koaCompose.default)([(0, _koaHelmet.default)(helmetOpts), (0, _koaBodyparser.default)(bodyParserOpts), (0, _cors.default)(), (0, _koaCompress.default)(compressOpts), (0, _dataCache.default)(dataCacheOpts), (0, _errorHandling.default)(errorHandlingOpts)]);
};

exports.default = _default;
